evm/hyperliquid.js

```diff
@@ -0,0 +1,225 @@
+const execAsync = promisify(exec);
+const msgpack = require('msgpack-lite');
+const { keccak256 } = require('ethers/lib/utils');
+
+function addressToBytes(address) {
+    return Buffer.from(address.replace('0x', ''), 'hex');
+}
+
+function actionHash(action, activePool, nonce) {
+    const actionData = msgpack.encode(action);
+    const nonceBuffer = Buffer.alloc(8);
+    nonceBuffer.writeBigUInt64BE(BigInt(nonce));
+
+    const vaultBuffer =
+        activePool === null || activePool === undefined
+            ? Buffer.from([0x00])
+            : Buffer.concat([Buffer.from([0x01]), addressToBytes(activePool)]);
+
+    const data = Buffer.concat([actionData, nonceBuffer, vaultBuffer]);
+    return keccak256(data);
+}
+
+function constructPhantomAgent(hash, source) {
+    return {
+        source: source,
+        connectionId: hash,
+    };
+}
+
+async function signL1Action(wallet, action, activePool, nonce, chain) {
+    const hash = actionHash(action, activePool, nonce);
+    const phantomAgent = constructPhantomAgent(hash, chain.hypercore?.source);
+    const domain = chain.hypercore?.domain;
+
+    const agent = [
+        { name: 'source', type: 'string' },
+        { name: 'connectionId', type: 'bytes32' },
+    ];
+
+    const signature = await wallet._signTypedData(domain, { Agent: agent }, phantomAgent);
+    const sig = ethers.utils.splitSignature(signature);
+
+    return { r: sig.r, s: sig.s, v: sig.v };
+}
+
+async function updateBlockSize(wallet, chain, options) {
+    const [blockSize] = options.args;
+    validateParameters({
+        isNonEmptyString: { blockSize },
+    });
+
+    const useBig = blockSize === 'big';
+    const network = chain.networkType;
+
+    printInfo('Block size', blockSize);
+    printInfo('Network', network);
+
+    const action = { type: 'evmUserModify', usingBigBlocks: useBig };
+    const nonce = Date.now();
+    const signature = await signL1Action(wallet, action, null, nonce, chain);
+    const payload = { action, signature, nonce };
+    const endpoint = `${chain.hypercore.url}/exchange`;
+    const result = await httpPost(endpoint, payload);
+
+    if (result.status !== 'ok') {
+        throw new Error(result);
+    }
+
+    printInfo('Result', result);
+    return result;
+}
+
+async function deployer(wallet, chain, options) {
+    const [tokenId] = options.args;
+    validateParameters({
+        isNonEmptyString: { tokenId },
+    });
+
+    const contracts = chain.contracts;
+    const interchainTokenFactoryAddress = contracts.InterchainTokenFactory?.address;
+    const interchainTokenServiceAddress = contracts.InterchainTokenService?.address;
+
+    validateParameters({
+        isValidAddress: { interchainTokenFactoryAddress, interchainTokenServiceAddress },
+    });
+
+    const IInterchainTokenService = getContractJSON('IInterchainTokenService');
+
+    const interchainTokenService = new Contract(interchainTokenServiceAddress, IInterchainTokenService.abi, wallet);
```

Comment:
```suggestion
    const interchainTokenService = new Contract(chain.contracts.InterchainTokenService?.address, IInterchainTokenService.abi, wallet);
```
remove single-use intermediate variables for brevity?